src/test/resources/com/puppycrawl/tools/checkstyle/checks/metrics/classfanoutcomplexity/InputClassFanOutComplexityLambdaNew.java

```diff
@@ -0,0 +1,40 @@
+/*
+ClassFanOutComplexity
+max = 0
+excludedClasses = (default)ArrayIndexOutOfBoundsException, ArrayList, Boolean, Byte, \
+                  Character, Class, Collection, Deprecated, Deque, Double, DoubleStream, \
+                  EnumSet, Exception, Float, FunctionalInterface, HashMap, HashSet, \
+                  IllegalArgumentException, IllegalStateException, IndexOutOfBoundsException, \
+                  IntStream, Integer, LinkedHashMap, LinkedHashSet, LinkedList, List, Long, \
+                  LongStream, Map, NullPointerException, Object, Optional, OptionalDouble, \
+                  OptionalInt, OptionalLong, Override, Queue, RuntimeException, SafeVarargs, \
+                  SecurityException, Set, Short, SortedMap, SortedSet, Stream, String, \
+                  StringBuffer, StringBuilder, SuppressWarnings, Throwable, TreeMap, TreeSet, \
+                  UnsupportedOperationException, Void, boolean, byte, char, double, float, \
+                  int, long, short, var, void
+excludedPackages = (default)
+
+
+*/
+
+package com.puppycrawl.tools.checkstyle.checks.metrics.classfanoutcomplexity;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+class InputClassFanOutComplexityLambdaNew {
+    public void testMethod(Map<String, List<String>> entry) {
+        List<Constraint> constraints = new ArrayList<>();
+        String string = "";
+        entry.values().stream()
+                .flatMap(List::stream)
+                .map(Constraint::new)
```

Comment:
`Constraint::new` will not be recognized as a reference and will not be recorded. I am adding this test file to pass CI right now. I think it would be better if we could fix this false negative in a separated PR to make everything easier.

Reply:
Please use here File class, as you did in issue description

Reply:
done